SpatialGDK/ExternalSchemaCodegen/Programs/CodeGenerator/Program.cs

```diff
@@ -0,0 +1,82 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using CommandLine;
+using Improbable.Codegen.Base;
+using Improbable.CodeGen.Base;
+using Improbable.CodeGen.Unreal;
+
+namespace CodeGenerator
+{
+    public class Options
+    {
+        [Option("input-bundle", Required = true,
+            HelpText = "The path to the JSON Bundle file output by the SpatialOS schema_compiler.")]
+        public string InputBundle { get; set; }
+
+        [Option("output-dir", Required = true,
+            HelpText = "The path to write the generated code to.")]
+        public string OutputDir { get; set; }
+    }
+
+    internal class Program
+    {
+        private static void Main(string[] args)
+        {
+            Parser.Default.ParseArguments<Options>(args)
+                .WithParsed(Run)
+                .WithNotParsed(errors =>
+                {
+                    foreach (var error in errors)
+                    {
+                        Console.Error.WriteLine(error);
+                    }
+
+                    Environment.ExitCode = 1;
+                });
+        }
+
+        private static void Run(Options options)
+        {
+            try
+            {
+                var bundle = SchemaBundleLoader.LoadBundle(options.InputBundle);
+
+                var generators = new List<ICodeGenerator>
+                {
+                    new UnrealGenerator()
+                };
+
+                var output = new List<GeneratedFile>();
+
+                generators.ForEach((ICodeGenerator g) => output.AddRange(g.GenerateFiles(bundle)));
+
+                foreach (var generatedFile in output)
+                {
+                    WriteFile(options, generatedFile.FilePath, generatedFile.Contents);
+                }
+            }
+            catch (Exception exception)
+            {
+                Debug.WriteLine(exception);
+                Console.Error.WriteLine(exception);
+                Environment.ExitCode = 1;
+            }
+        }
+
+        private static void WriteFile(Options options, string filename, string text)
```

Comment:
This function is not a good abstraction because it takes a whole `Options` instance but uses just one member, takes a `filename` but it looks like a file path is passed in the call above. Wouldn't it be clearer if it took a string for the output directory path and a `GeneratedFile` as a second parameter?

Reply:
agreed